staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -588,8 +588,8 @@ func (c *Cacher) Get(ctx context.Context, key string, resourceVersion string, ob
 func (c *Cacher) GetToList(ctx context.Context, key string, resourceVersion string, pred storage.SelectionPredicate, listObj runtime.Object) error {
 	pagingEnabled := utilfeature.DefaultFeatureGate.Enabled(features.APIListChunking)
 	hasContinuation := pagingEnabled && len(pred.Continue) > 0
-	hasLimit := pagingEnabled && pred.Limit > 0 && resourceVersion != "0"
-	if resourceVersion == "" || hasContinuation || hasLimit {
+	hasLimit := pagingEnabled && pred.Limit > 0 && resourceVersion != "0" && resourceVersion != ""
```

Comment:
please don't change the construction of `hasLimit`... this is unlikely to be backwards compatible. if there's an additional condition we need to check, separate that into a different variable

Reply:
Sure, I will update code according to this kubernetes/enhancements#1404.

Reply:
@liggit fixed.